### PR TITLE
[ONNX] Add converter for QAttention from Microsoft onnxruntime contrib opset

### PR DESCRIPTION
**Pull request by KJlaccHoeUM9l**
This PR adds support for [QAttention](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.QAttention) - quantized version of [Attention](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#com.microsoft.Attention) from Microsoft onnxruntime contrib opset.
An explanation and illustration of how this layer works can be found, for example, in @lena-voita [NLP course](https://lena-voita.github.io/nlp_course/seq2seq_and_attention.html).

### CODE REVIEW

**Comment by KJlaccHoeUM9l** [comment]
Hello @vvchernov, @echuraev, @AndrewZhaoLuo!
Could you review this PR?

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -1379,6 +1379,298 @@ def massage(tensor):
...
+        #       1. (batch_size)
+        #       2. (2 * batch_size)
+        #       3. (batch_size, 1)
+        #       4. (1, 1)
+        #       5. (batch_size, past_sequence_length + sequence_length)
+        #  In practice, for GPT-2 there shape is (batch, past_seq_length + seq_length).
+        #  Currently only (batch, past_seq_length + seq_length) shape is supported.
+        mask_index = inputs[5]
+
+        # Scalar, which means a per-tensor/layer quantization
```
Comment (echuraev) [inline python/tvm/relay/frontend/onnx.py:1423]: nit: You have absolutely the same comment for `input[3]` and `input[7]`

File: python/tvm/relay/frontend/onnx.py [hunk 2]
```diff
@@ -1379,6 +1379,298 @@ def massage(tensor):
...
+        ):
+            rhs_transposed = _op.transpose(rhs, axes=[0, 2, 1])  # QNN Batch Matmul do: X * Y^T
+            result = _qnn.op.batch_matmul(
+                lhs, rhs_transposed, lhs_zero_point, rhs_zero_point, lhs_scale, rhs_scale
+            )
+            result = _qnn.op.dequantize(
+                result,
+                _op.multiply(lhs_scale, rhs_scale),
+                zero_point_zero,
+                axis=-1,  # TODO(agladyshev): what is 'axis' parameter for?
```
Comment (echuraev) [inline python/tvm/relay/frontend/onnx.py:1556]: Do you still need this todo comment?

**Review by echuraev — COMMENTED** [review]
LGTM. But I don't have a lot of knowledge in this codebase. @jwfromm, @AndrewZhaoLuo could you please take a look at this PR?

**Comment by AndrewZhaoLuo** [comment]
Apologies, I've been quite sick. I'll try to look at this Thursday.

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (KJlaccHoeUM9l) [reply python/tvm/relay/frontend/onnx.py:1426]: done

File: python/tvm/relay/frontend/onnx.py (hunk 2 above)
Comment (KJlaccHoeUM9l) [reply python/tvm/relay/frontend/onnx.py]: done

**Review by AndrewZhaoLuo — APPROVED** [review]
Thanks! Sorry for late review. LGTM

**Pull request merged**